sbncode/CAFMaker/CAFMaker_module.cc

```diff
@@ -1280,11 +1278,17 @@ void CAFMaker::produce(art::Event& evt) noexcept {
         thisShower = fmShower.at(iPart);
       }
 
-      if (!thisTrack.empty())  { // it's a track!
+      assert(thisParticle.size() == 1);
+     
+      rec.reco.pfp.push_back(SRPFP());
+      const larpandoraobj::PFParticleMetadata *pfpMeta = (fmPFPMeta.at(iPart).empty()) ? NULL : fmPFPMeta.at(iPart).at(0).get();
+      FillPFPVars(thisParticle, primary, pfpMeta, rec.reco.pfp.back());
+
+      rec.reco.pfp.back().trk = SRTrack();
+      rec.reco.pfp.back().shw = SRShower();
```

Comment:
This line should be unnecessary, as `shw` constructor should have already been called when `SRPFP` was.
```suggestion
```